src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,27 @@
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Error adding movie", e);
+        }
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            Movie movie = session.get(Movie.class, id);
+            return Optional.ofNullable(movie);
+        } catch (Exception e) {
+            throw new DataProcessingException("Error retrieving movie", e);
```

Comment:
For the `get` method, it would be beneficial to include the movie ID in the exception message to provide more context and aid in debugging if an error occurs.